anton_potapov/graph.hpp

```diff
@@ -243,16 +244,67 @@ class Graph {
         }
       }
     }
-    size_t new_max_depth = 0;
+    int new_max_depth = 0;
     for (const auto& [vertex_id, depth] : depths) {
       vertices_.find(vertex_id)->second.depth = depth;
       if (depth > new_max_depth) {
         new_max_depth = depth;
       }
     }
-    max_depth_ = new_max_depth;
     for (const auto& [vertex_id, depth] : depths) {
       vertices_at_depth_[depth].insert(vertex_id);
     }
   }
+
+ private:
+  VertexId next_vertex_id_{};
+  EdgeId next_edge_id_{};
+  std::map<VertexId, Vertex> vertices_;
+  std::map<EdgeId, Edge> edges_;
+  std::map<int, std::set<VertexId>> vertices_at_depth_;
+
+  const Vertex& get_vertex(const VertexId& id) const {
+    return vertices_.find(id)->second;
+  }
+  const Edge& get_edge(const EdgeId& id) const {
+    return edges_.find(id)->second;
+  }
+
+  VertexId get_next_vertex_id() { return next_vertex_id_++; }
+
+  EdgeId get_next_edge_id() { return next_edge_id_++; }
+
+  std::string get_json_string_private() const {
+    std::stringstream json_stringstream;
+    json_stringstream << "{\"depth\":" << this->max_depth() << ",";
+    json_stringstream << "\"vertices\":[";
+    for (auto it = vertices_.begin(); it != vertices_.end(); ++it) {
+      json_stringstream << it->second.get_json_string();
+      if (std::next(it) != vertices_.end()) {
+        json_stringstream << ",";
+      }
+    }
+    json_stringstream << "],\"edges\":[";
+    for (auto it = edges_.begin(); it != edges_.end(); ++it) {
+      json_stringstream << it->second.get_json_string();
+      if (std::next(it) != edges_.end()) {
+        json_stringstream << ",";
+      }
+    }
+    json_stringstream << "]}";
+    return json_stringstream.str();
+  }
 };
+
+VertexId get_random_vertex_id(const std::set<VertexId>& vertex_id_set) {
+  int range_start = 0;
+  int range_end = (int)vertex_id_set.size() - 1;
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<> dist(range_start, range_end);
+  size_t rand_int = dist(gen);
+  size_t rand_id = rand_int % vertex_id_set.size();
+  auto rand_it = vertex_id_set.begin();
+  std::advance(rand_it, rand_id);
+  return *rand_it;
```

Comment:
Так не попроще будет выглядеть?
```suggestion
  std::random_device rd;
  std::mt19937 gen(rd());
  std::uniform_int_distribution<> dist(0, vertex_id_set.size() - 1);
  auto iter = vertex_id_set.begin() + dist(gen);
  return *iter;
```

Reply:
92e638eaf9cc58cfab888e31bbfde8a6e2069147 сделал, только у меня там не random access итераторы